src/V3Clock.cpp

```diff
@@ -120,6 +121,27 @@ class ClockVisitor final : public VNVisitor {
         UINFO(4, "New Last: " << newvscp << endl);
         return newvscp;
     }
+    AstVarScope* getCreateSampledVar(AstVarScope* vscp) {
+        if (vscp->user1p()) return static_cast<AstVarScope*>(vscp->user1p());
+        const AstVar* const varp = vscp->varp();
+        const string newvarname
+            = (string("__Vsampled__") + vscp->scopep()->nameDotless() + "__" + varp->name());
+        AstVar* const newvarp
+            = new AstVar(vscp->fileline(), VVarType::MODULETEMP, newvarname, varp->dtypep());
+        newvarp->noReset(true);  // Reset by below assign
+        m_modp->addStmtp(newvarp);
+        AstVarScope* const newvscp = new AstVarScope(vscp->fileline(), m_scopep, newvarp);
+        vscp->user1p(newvscp);
+        m_scopep->addVarp(newvscp);
+        // At top, assign them
+        AstAssign* const finalp = new AstAssign(
+            vscp->fileline(), new AstVarRef(vscp->fileline(), newvscp, VAccess::WRITE),
+            new AstVarRef(vscp->fileline(), vscp, VAccess::READ));
+        m_evalFuncp->addInitsp(finalp);
+        //
```

Comment:
Do we need that?

Reply:
I don't think so, but it's placed here because it occurs in a corresponding place in the function used for clocks.